From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Fri, 11 Jul 2025 08:26:10 +0200
Subject: [PATCH] Drop timestamp in generated source code

Fixes #72206

This helps reproducible builds of libomp.so
probably because LLVM's LTO computed a hash of inputs
to generate its symbol names.

This patch was done while working on reproducible builds for openSUSE.
---
 
Index: openmp-15.0.7.src/docs/conf.py
===================================================================
--- a/projects/openmp/docs/conf.py
+++ b/projects/openmp/docs/conf.py
@@ -12,7 +12,6 @@
 # serve to show the default.
 
 import sys, os
-from datetime import date
 
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
@@ -42,7 +41,7 @@ master_doc = "index"
 
 # General information about the project.
 project = "LLVM/OpenMP"
-copyright = "2013-%d, LLVM/OpenMP" % date.today().year
+copyright = "2013-%d, LLVM/OpenMP" % 2023
 
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
Index: openmp-15.0.7.src/runtime/tools/message-converter.pl
===================================================================
--- a/projects/openmp/runtime/tools/message-converter.pl
+++ b/projects/openmp/runtime/tools/message-converter.pl
@@ -50,7 +50,7 @@ sub _generate_comment($$$) {
     my $bulk =
         $open . " Do not edit this file! " . $close . "\n" .
         $open . " The file was generated from " . get_file( $data->{ "%meta" }->{ source } ) .
-            " by " . $tool . " on " . localtime() . ". " . $close . "\n";
+            " by " . $tool . ". " . $close . "\n";
     return $bulk;
 
 }; # sub _generate_comment
